### PR TITLE
#159568537 Enable articles to be paid for

### PR DESCRIPTION
**Pull request by Oloyedesinmiloluwa**
#### What does this PR do?
Add the feature to enable articles to be paid for
#### Description of Task to be completed?
    - update create article methods to set price
    - update, update article controllers
    - update model and migration files
    - add validation methods for price input
    - add unit tests for associated edge cases
#### How should this be manually tested?
Run these commands;
- npm install
- npm test
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/159568537

### CODE REVIEW

File: middlewares/validatePrice.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const validatePrice = (req, res, next) => {
+  if (req.body.article.price && typeof req.body.article.price !== 'number') {
```
[the remark this answers is not in the document]
Comment (Oloyedesinmiloluwa) [reply middlewares/validatePrice.js]: Fixed

**Review by ascii-dev — APPROVED** [review]
Good job...

**Review by klevamane — APPROVED** [review]
Ok

File: helpers/createArticleHelper.js [hunk 2]
```diff
@@ -11,8 +11,9 @@ import { Article, User } from '../models';
  */
 
 const createArticleHelper = (res, articleObject, imageUrl = null) => {
+  articleObject.price = (articleObject.price) ? articleObject.price.toFixed(2) : undefined;
```
Comment (anuonifade) [inline helpers/createArticleHelper.js]: why make this `undefined` rather than 0?

File: middlewares/validatePrice.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const validatePrice = (req, res, next) => {
+  const maxPrice = 0.28;
```
Comment (andela-aonifade) [inline middlewares/validatePrice.js]: Why this line?

File: middlewares/validatePrice.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const minPrice = 5.53;
```
Comment (andela-aonifade) [inline middlewares/validatePrice.js]: And this?

**Comment by anuonifade** [comment]
@Oloyedesinmiloluwa please update your PR and address my comments.

File: helpers/createArticleHelper.js (hunk 2 above)
Comment (Oloyedesinmiloluwa) [reply helpers/createArticleHelper.js]: My approach before was to leave it as `undefined` if it is not defined, so far `isPaidFor` is not set to `true` but I realize now that making it `0` makes better sense. I will fix it, thanks

**Review by anuonifade — APPROVED** [review]
OK

**Review by anuonifade — APPROVED** [review]
OK
